src/game_battlealgorithm.cpp

```diff
@@ -639,7 +639,8 @@ void Game_BattleAlgorithm::Normal::Apply() {
 
 	source->SetCharged(false);
 	if (source->GetType() == Game_Battler::Type_Ally && static_cast<Game_Actor*>(source)->GetWeaponId() != 0) {
-		source->ChangeSp(-source->CalculateSkillCost(static_cast<Game_Actor*>(source)->GetWeaponId()));
+		Game_Actor* src = static_cast<Game_Actor*>(source);
```

Comment:
I think you can move the cast two lines up an use `src` already in the condition.